rust/agents/relayer/src/msg/metadata/aggregation.rs

```diff
@@ -140,21 +147,34 @@ impl MetadataBuilder for AggregationIsmMetadataBuilder {
                 .map(|ism_address| self.base.build_ism(*ism_address)),
         )
         .await;
-
-        let filtered_sub_module_metas = metas
+        let (valid_sub_modules, invalid_sub_modules): (Vec<_>, Vec<_>) = metas
             .into_iter()
             .enumerate()
             .zip(sub_modules.into_iter())
-            .filter_map(|((index, meta_result), sub_module_result)| {
-                match (meta_result, sub_module_result) {
-                    (Ok(Some(meta)), Ok(ism)) => Some(IsmAndMetadata::new(ism, index, meta)),
-                    _ => None,
-                }
+            .partition(|&((_, ref meta_result), ref sub_module_result)| {
```

Comment:
i don't understand what these changes are

Reply:
right now the aggregation log is "could not find all metadata: found 1 out of 2 required ISMs" which doesn't tell which ism is not working so the partition separates the working and missing isms which helps to log which ISMs aren't working in case metas_and_gas_count < threshold